rpcs3/Input/hid_pad_handler.cpp

```diff
@@ -104,6 +107,9 @@ hid_device* HidDevice::open()
 		hidDevice = hid_open_path(path.c_str());
 	}, false);
 #else
+	// Lock before calling "hid_open_path()"
+	std::unique_lock static_lock(g_hid_mutex, std::defer_lock);
```

Comment:
That makes no sense, it's dead code.
Just move the static_lock stuff in the mac path to the beginning of the function.
btw, just rename it to "lock", since it's not static (probably was in some intermediate commits of mine)

Reply:
ops, yes.
Do you mean, this:

```
hid_device* HidDevice::open()
{
	// Lock before calling "hid_open_path()"
	std::unique_lock lock(g_hid_mutex, std::defer_lock);

	if (!lock.try_lock())
	{
		// The enumeration thread is busy. If we lock and open the device, we might get input stutter on other devices.
		return nullptr;
	}

#ifdef ANDROID
	hidDevice = hid_libusb_wrap_sys_device(path, -1);
#elif defined(__APPLE__)
	Emu.BlockingCallFromMainThread([this]()
	{
		hidDevice = hid_open_path(path.c_str());
	}, false);
#else
	hidDevice = hid_open_path(path.data());
#endif

	return hidDevice;
}

```

Reply:
yes